File: AspNetMvcUnleashed/0672329980_source/Chapter07/VB/MvcApplication1/MvcApplication1/Controllers/WidgetController.vb
Public Class WidgetController
    Inherits System.Web.Mvc.Controller


    ' GET: /Widget/Create
    Function Create() As ActionResult
        Return View()
    End Function


    ' POST: /Widget/Create
    <AcceptVerbs(HttpVerbs.Post)> _
    Function Create(ByVal widgetToCreate As Widget) As ActionResult
        If Not ModelState.IsValid Then
            Return View()
        End If

        ' Add widget to database
        Return RedirectToAction("Index")
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter07/CS/MvcApplication1/MvcApplication1/Controllers/WidgetController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Web.Mvc.Ajax;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class WidgetController : Controller
    {

        //
        // GET: /Widget/Create

        public ActionResult Create()
        {
            return View();
        } 

        //
        // POST: /Widget/Create

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create(Widget widgetToCreate)
        {
            if (!ModelState.IsValid)
                return View();

            // Insert widget into database

            return RedirectToAction("Index");
        }


    }
}
